lib/ffi-icu/time_formatting.rb

```diff
@@ -215,6 +232,93 @@ def skeleton_format(skeleton_pattern_str, locale)
           retry
         end
       end
+
+      private
+
+      # Converts the current pattern to a pattern that takes the desired hour cycle
+      # into account. This is needed because most of the standard patterns in ICU
+      # contain either h (12 hour) or H (23 hour) in them, instead of j (locale-
+      # specified hour cycle). This means if you use a locale with an @hours=h12
+      # keyword in it, for example, it would normally be totally ignored by ICU.
+      #
+      # This is the same fixup done by Firefox:
+      # https://github.com/tc39/ecma402/issues/665#issuecomment-1084833809
+      # https://searchfox.org/mozilla-central/rev/625c3d0c8ae46502aed83f33bd530cb93e926e9f/intl/components/src/DateTimeFormat.cpp#282-323
+      def fixup_pattern_hour_cycle!
```

Comment:
Since it's very similar, what do you think about using a similar method name [ReplaceHourSymbol](https://searchfox.org/mozilla-central/source/intl/components/src/DateTimeFormat.cpp#136)?

Reply:
Is your suggestion to simply rename the method (no problem) or to factor out the skeleton and pattern munging into a single method called either on a pattern or skeleton? I’m reluctant to do the latter because there are a few subtle differences in eg whether am/pm markers should be removed, and whether j locale-appropriate hour markers can be added

Reply:
Just suggesting changing the method name to be similar to Intl.